### DISCUSSION TITLE
HA access stream remotely and add a camera entity?

### DISCUSSION THREAD

**Discussion by N3rd7** [mrlt8/docker-wyze-bridge#1339] [Q&A] [answered]
Hi all,
Fantastic add-on, thank you mrlt8! The Docker wyze bridge works great as add-on to HA (running on raspi4b), weather I am home or remote. However I'd like to add a video stream to specific camera in specific area in my dashboard. Since the cameras don't show as camera entities in my HA, I added them using the webpage card and entering my homeassistant.local:PORT/camera-name. It works when I am connected to HA directly on my network, but if I connect via nabu casa (https connection), it will not show the live stream with this message in place of the stream on the dashboard: "unable to load iframe pointing at website using http: if home assistant is served over https".
Is there a way to make live streams available remotely? I saw somewhere in the documentation that cameras might be added as entities using the generic camera service of HA, but I just can't figure out how to add a camera using this. 
Any help would be great! Thanks :)

**Comment by N3rd7** [comment, marked as answer]
Nevermind, I just found how to use generic camera. I am amazed (scared?) that the system didn't tag me as fraudulent activity after a million attempts to log in... But it works now, so I'm happy
If anyone comes across this post looking for an answer, here is what I did:
1. Add generic camera integration
2. In the configuration:
- Stream URL is rtsp://127.0.0.1:8554/cam-name (obviously, customize cam-name: all lower case, replace all spaces in your wyze cam name with a dash -)
- RTSP transport protocol: select TCP
- Authentication: leave as basic
- User name: it is wb by default, unless you customized the stream authentication username and password
- Password: default is your custom API key that shows at the bottom of the HA Docker Wyze Bridge page

I guess it's not that hard, it just took me a while to understand which protocol to select and authentication I needed to use...

**Reply by Geczy** [reply]
did you fill out the still image url field? i'm having trouble with that one

thank you for the auth details, i had no idea we needed that

**Reply by Geczy** [reply]
this worked: https://github.com/mrlt8/docker-wyze-bridge/issues/1199#issuecomment-2112870446

**Reply by G2740** [reply]
Thanks for the tips. I agree it's hard, for me anyhow. I'll give this a shot tomorrow, it took me a while to find it, but thankfully I did.🤞👍

**Reply by N3rd7** [reply]
Hi G2740, apologies I totally missed your question! No I did not fill the still image URL. only the live stream URL

**Reply by G2740** [reply]
> Hi G2740, apologies I totally missed your question! No I did not fill the still image URL. only the live stream URL

That was another post question, no biggie 😉. I'll further help document this procedure some, once I figure it out myself, lol and if I remember what I did. Lol And post it around to be found.

**Reply by l00nix** [reply]
Has this changed? I am trying the instructions provided above but it times out all the time

![image](https://github.com/user-attachments/assets/36f81fbc-e98a-4fe6-9064-f6d3d6098040)

I tried with and without 'Verify SSL certificate' enabled.

Wyze Bride HA plugin works

![image](https://github.com/user-attachments/assets/eb754c8b-7aac-408b-8ec0-346e508f9c44)

Any ideas?

Thanks

Alex

**Reply by G2740** [reply]
Don't remember for sure, try clicking stream and copy the rtsp path from there and paste it(I think).

Presumes you have a Wyze key and ID also.

Then setup generic camera in HA to display. I don't have it loaded any longer. Good luck!

**Reply by N3rd7** [reply]
username should be wb, not wp :) Hopefully this is the only reason why it's not working! Also make sure the password is your entire DWB API key, with no space at the end

**Reply by l00nix** [reply]
@N3rd7 - how embarrassing ;-) lol I tried with username wb but am now experiencing this issue https://github.com/mrlt8/docker-wyze-bridge/issues/1459. Thanks

**Reply by N3rd7** [reply]
Oh so Wyze Bridge doesn't work anymore? 
Just to be sure: have you switched to an updated fork (for HAOS users, I recommend jdeath's fork) https://github.com/jdeath/docker-wyze-bridge. 
If not, here are step by steps to replace your current add-on with the updated one (Maybe overkill in details, but I don't know the level of guidance you would want, so I went "all-in" :))

- Go to your current HAOS settings/add-ons, open docker wyze bridge (mrtl8 version I assume)
- Open the configuration tab, click on the 3 dots in the top right corner, and select "edit in yaml"
- Copy the entire code and paste it somewhere to save the code (temporary, you will only need to paste it once)
- Back to the DWB add-on info tab, select stop then uninstall
- Back on the settings/add-on page, click bottom right "Add-on store". At this point, you shouldn't see Docker Wyze bridge in the list anymore, since you uninstalled it. If you still see it, try restarting HA. 
- Click on the 3 dots in the top right corner, and select "Repositories"
- In the box, paste the jdeath fork address (https://github.com/jdeath/docker-wyze-bridge) and click "Add"
- Docker Wyze Bridge should now show again in the list of add-ons in the store. Open the first tile called "Docker Wyze Bridge" (nothing else in the title), and click install
- Once installed, turn on "start on boot" if not on already, then open the configuration tab, click on the 3 dots in the top right corner, and edit in yaml. Select the entire code, delete, and paste the code you saved earlier.
- If you didn't have this in your previous setup, I recommend turning off "On Demand" (Add _ON_DEMAND: false_ at the beginning of the configuration code)
- Restart, DWB should now work for you (fingers crossed!)
